torch/distributed/distributed_c10d.py

```diff
@@ -1156,6 +1157,148 @@ def all_gather_multigpu(output_tensor_lists,
         work.wait()
 
 
+def _object_to_tensor(obj):
+    buffer = io.BytesIO()
+    torch.save(obj, buffer)
```

Comment:
I see, I went with `torch.save`/`torch.load` since it seems that is the de-facto way of doing serialization in PT. I think it might be more efficient to just use the `pickle` module directly, though.